crates/cards/src/validate.rs

```diff
@@ -22,6 +22,9 @@ impl CardNumber {
     pub fn get_card_isin(self) -> String {
         self.0.peek().chars().take(6).collect::<String>()
     }
+    pub fn get_last4(self) -> String {
+        self.0.peek().chars().rev().take(4).collect::<String>()
```

Comment:
If your last 4 digits are `1234`, this would return `4321`, isn't it?

Adding a `rev()` after the `take()` should provide us the intended result.

```suggestion
        self.0.peek().chars().rev().take(4).rev().collect::<String>()
```

Reply:
self.0.peek().chars().rev().take(4).rev().collect::<String>()
   |                                                   ^^^^^^^ method cannot be called on `Rev<Take<Rev<Chars<'_>>>>` due to unsatisfied trait bounds
   |
   
   i am getting this error, if i try to rev() after take(4)

Reply:
In that case, please revert to your original implementation with the intermediate `collect()`.

Reply:
done!